addons/sale_timesheet/controllers/portal.py

```diff
@@ -115,8 +115,9 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
         except (AccessError, MissingError):
             pass
 
-        if task.sale_order_id.invoice_ids:
-            moves = request.env['account.move'].search([('id', 'in', task.sale_order_id.invoice_ids.ids)])
+        moves = task.sale_order_id.invoice_ids
+        if moves and request.env['account.move'].check_access_rights('read', raise_exception=False):
+            moves = request.env['account.move'].search([('id', 'in', moves.ids)])
```

Comment:
I think we cannot access to `task.sale_order_id.invoice_ids` if `request.env['account.move'].check_access_rights('read', raise_exception=False)` is False.

Can you check?

Reply:
@xavierbol , we can access it because the `task` passed in this method is accessed with sudo[1] so it will check this statement with sudo.
and due to that we are doing the search again like: `request.env['account.move'].search([('id', 'in', moves.ids)])`
instead of directly accessing the `task.sale_order_id.invoice_ids`

Thanks
[1]: https://github.com/odoo/odoo/blob/16.0/addons/project/controllers/portal.py#L190